bot/commands/exchange.py

```diff
@@ -0,0 +1,64 @@
+import logging
+
+from telegram import Update
+from telegram.ext import CommandHandler, CallbackContext
+
+from bot.core import BotTelegramCore
+from bot.utils import convert_dcr
+from bot.exceptions import DcrDataAPIError, ExchangeAPIError
+
+
+logging.basicConfig(
+    format='%(asctime)s - %(name)s - %(levelname)s - %(message)s',
+    level=logging.INFO)
+
+logger = logging.getLogger(__name__)
+
+
+def dcr(update: Update, context: CallbackContext):
+    target_currency = None
+    dcr_amount = None
+
+    try:
+        arg_0 = context.args[0]
+        if arg_0.isdigit():
+            dcr_amount = float(arg_0)
+        else:
+            target_currency = arg_0.upper()
+    except IndexError:
+        dcr_amount = 1
+        target_currency = "USD"
+
+    try:
+        arg_1 = context.args[1]
+        if arg_1.isdigit():
+            dcr_amount = float(arg_1)
+        else:
+            target_currency = arg_1.upper()
+    except IndexError:
+        if target_currency is None:
+            target_currency = "USD"
+        if dcr_amount is None:
+            dcr_amount = 1
```

Comment:
This is not the same thing? except IndexError

Reply:
On the first `try` block I'm looking for the first arg. It may be a string amout `3.5` or a string currency `usd`. So, if there was a first arg we have either `target_currency` or `dcr_amount` assigned.

On second `try` block I'm looking for the second arg. If there wasn't the second arg, we can't overwrite the first assigned value (either `target_ currency` or `dcr_amount`). Thats why I did this `if` checking if a value is `None` before reassign.

## EG
### User called `/dcr 3`
On first try we have
- `dcr_amount = 3`
- `target_currency = None`

On secont try
- we don't overwrite `dcr_amount`
- assign `target_currency = 'usd'`

### User called `/dcr usd 3`
On first try we have
- `dcr_amount = None`
- `target_currency = 'usd'`

On secont try
- assign `dcr_amount = 3`
- we don't overwrite `target_currency`